pkg/tfshim/sdk-v2/provider2.go

```diff
@@ -262,13 +267,20 @@ func (p *planResourceChangeImpl) Diff(
 		return nil, err
 	}
 
+	//nolint:lll
+	// https://github.com/opentofu/opentofu/blob/864aa9d1d629090cfc4ddf9fdd344d34dee9793e/internal/tofu/node_resource_abstract_instance.go#L1024
+	unmarkedPrior, _ := st.UnmarkDeep()
```

Comment:
contract.AssertNoError(err) on the errors?

Reply:
The second value is not errors but the list of marks - we don't need that